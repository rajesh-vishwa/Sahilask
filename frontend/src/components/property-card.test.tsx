import React from "react";

import PropertyCard from "./property-card.component";

import { shallow } from "enzyme";
import { IProperty } from "../models/property.model";

describe("<PropertyCard />", () => {
  const propertyProps: IProperty = {
    id: 1,
    name: "Purple Bricks",
    image: "/images/airpods.jpg",
    numberOfBedrooms: 2,
    address: "Mansford Street, London E2",
    postcode: "E2Y00",
    description: "2 rooms flat for sale",
    expired: false,
  };
  const wrapper = shallow(<PropertyCard property={propertyProps} />);
  it("should match the snapshot ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should accepts property props", () => {
    expect(wrapper.props().property).not.toBeNull();
  });
});
