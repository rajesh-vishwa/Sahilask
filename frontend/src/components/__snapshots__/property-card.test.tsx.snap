// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyCard /> should match the snapshot  1`] = `
<Card
  body={false}
  className="my-3 p-3 rounded"
>
  <Row
    noGutters={false}
  >
    <Col>
      <CardImg
        src="/images/airpods.jpg"
        variant={null}
      />
    </Col>
    <Col>
      <CardBody>
        <CardTitle
          as="div"
        >
          <h1>
            Purple Bricks
          </h1>
        </CardTitle>
        <CardText
          as="div"
        >
          Number of bed: 
          2
        </CardText>
        <CardText
          as="div"
        >
          Address: 
          Mansford Street, London E2
          , 
          E2Y00
        </CardText>
      </CardBody>
      <CardBody>
        <CardText
          as="div"
        >
          <div
            className="my-3"
          >
            <h3>
              2 rooms flat for sale
            </h3>
          </div>
        </CardText>
        <CardText
          as="div"
        >
          <h3>
            Is Expred? 
            No
          </h3>
        </CardText>
      </CardBody>
    </Col>
  </Row>
</Card>
`;
